blocks/hooks/anchor.js

```diff
@@ -59,21 +59,25 @@ export function addAttribute( settings ) {
 export const withInspectorControl = createHigherOrderComponent( ( BlockEdit ) => {
 	return ( props ) => {
 		const hasAnchor = hasBlockSupport( props.name, 'anchor' ) && props.isSelected;
-		return [
-			<BlockEdit key="block-edit-anchor" { ...props } />,
-			hasAnchor && <InspectorControls key="inspector-anchor">
-				<TextControl
-					label={ __( 'HTML Anchor' ) }
-					help={ __( 'Anchors lets you link directly to a section on a page.' ) }
-					value={ props.attributes.anchor || '' }
-					onChange={ ( nextValue ) => {
-						nextValue = nextValue.replace( ANCHOR_REGEX, '-' );
-						props.setAttributes( {
-							anchor: nextValue,
-						} );
-					} } />
-			</InspectorControls>,
-		];
+		return (
+			<Fragment>
+				<BlockEdit { ...props } />
+				{ hasAnchor && (
```

Comment:
Minor: Given that we are changing this zone, hasAnchor maybe confusing has it includes the isSelected condition. Maybe we can have a condition that checks this `hasBlockSupport( props.name, 'anchor' ) && props.isSelected` and if true returns what the fragment we have without other checks.
Otherwise we just return <BlockEdit { ...props } />.

Reply:
Updated, good idea. Thanks 👍